### PR TITLE
Added support for AdoptOpenJDK

### PR DESCRIPTION
**Pull request by breucode**
Docs need to be updated.

I would add them, when you approve of this mr, and then you can merge it :).

### CODE REVIEW

**Comment by breucode** [comment]
CI tests are also missing

File: .travis.yml [hunk 1]
```diff
@@ -17,6 +17,12 @@ env:
   - EXPECTED_VERSION=10.0.2 JDKW_DIST=zulu JDKW_BUILD=10.3+5 JDKW_VERSION=10.0.2 JDKW_VERBOSE=1
   - EXPECTED_VERSION=11.0.3 JDKW_DIST=zulu JDKW_BUILD=11.31.11 JDKW_VERSION=11.0.3 JDKW_VERBOSE=1
   - EXPECTED_VERSION=12.0.1 JDKW_DIST=zulu JDKW_BUILD=12.2.3 JDKW_VERSION=12.0.1 JDKW_VERBOSE=1
+  - EXPECTED_VERSION=1.8.0_232 JDKW_DIST=adpt JDKW_BUILD=b09 JDKW_VERSION=8u232 JVM_IMPL=hotspot JDKW_VERBOSE=1
```
Comment (vjkoskela) [inline .travis.yml]: Unless `adpt` is a common short-hand for Adopt JDK, I would prefer we use `adopt` as the identifier and reflect that in the code as well.

File: examples/adpt.hotspot.11.jdkw [hunk 2]
```diff
@@ -0,0 +1,6 @@
+JDKW_RELEASE=latest
+JDKW_DIST=adpt
+JDKW_BUILD=10
+JDKW_VERSION=11.0.5
+JVM_IMPL=hotspot
```
Comment (vjkoskela) [inline examples/adpt.hotspot.11.jdkw]: This new flag will need to be documented.

File: jdkw-impl.sh [hunk 3]
```diff
@@ -291,12 +312,18 @@ eval "${cmd_configuration}"
...
 dist_oracle="oracle"
 dist_zulu="zulu"
+dist_adpt="adpt"
 libc_musl="musl"
 libc_glibc="glibc"
 platform_macosx=""
 platform_linux=""
 platform_solaris=""
 platform_windows=""
+os_macosx="mac"
```
[the remark this answers is not in the document]
Comment (breucode) [reply jdkw-impl.sh]: I removed this and used jdkw_platform instead

File: .travis.yml (hunk 1 above)
Comment (breucode) [reply .travis.yml]: Changed every occurence of adpt to adopt

File: jdkw-impl.sh [hunk 4]
```diff
@@ -435,9 +480,18 @@ if [ "${JDKW_DIST}" = "${dist_oracle}" ]; then
...
   alternate_jdk_source='http://cdn.azul.com/zulu/bin/zulu${JDKW_BUILD}-ca-jdk${JDKW_VERSION}-${JDKW_PLATFORM}.${JDKW_EXTENSION}'
+elif [ "${JDKW_DIST}" = "${dist_adpt}" ]; then
+  if [ "${java_major_version}" = "8" ]; then
+    release_name="jdk${JDKW_VERSION}-${JDKW_BUILD}"
+  else
+    release_name="jdk-${JDKW_VERSION}+${JDKW_BUILD}"
+  fi
+
+  primary_jdk_source='https://api.adoptopenjdk.net/v3/binary/version/${release_name}/${JDKW_OS}/${JDKW_PLATFORM}/jdk/${JVM_IMPL}/normal/adoptopenjdk'
+  alternate_jdk_source='https://api.adoptopenjdk.net/v3/binary/version/${release_name}/${JDKW_OS}/${JDKW_PLATFORM}/jdk/${JVM_IMPL}/normal/adoptopenjdk'
```
[the remark this answers is not in the document]
Comment (breucode) [reply jdkw-impl.sh]: Removed the alternate_jdk_source

File: examples/adpt.hotspot.11.jdkw (hunk 2 above)
Comment (breucode) [reply examples/adpt.hotspot.11.jdkw]: Added docs

**Comment by breucode** [comment]
Thank you for the detailed feedback! I changed all the stuff you mentioned.

File: jdkw-impl.sh [hunk 5]
```diff
@@ -422,7 +459,7 @@ fi
 
 # Special rules
 if [ "${JDKW_PLATFORM}" = "${platform_macosx}" ] && [ "${JDKW_DIST}" = "${dist_oracle}" ] && [ "${java_major_version}" = "6" ] ; then
-  log_err "JDK${java_major_version} from ${dist_oracle} is not supported on ${platform_macosx}"
+  log_err "JDK${java_major_version} from ${dist_oracle} is not supported on ${os_macosx}_${platform_macosx}"
```
Comment (vjkoskela) [inline jdkw-impl.sh]: I believe that `${os_macosx}` is no longer needed here.

File: README.md [hunk 6]
```diff
@@ -52,6 +52,7 @@ oracle       | `9.0.1`             | [sample](https://raw.githubusercontent.com/
 oracle       | `9.0.4` to current  | [sample](https://raw.githubusercontent.com/KoskiLabs/jdk-wrapper/master/examples/oracle.9.0.4-current.jdkw)
 oracle       | `10.0.1`            | [sample](https://raw.githubusercontent.com/KoskiLabs/jdk-wrapper/master/examples/oracle.10.0.1-current.jdkw)
 zulu         | any available       | [sample](https://raw.githubusercontent.com/KoskiLabs/jdk-wrapper/master/examples/zulu.any.jdkw)
+adopt        | [any of these](https://api.adoptopenjdk.net/v3/info/release_names?release_type=ga&vendor=adoptopenjdk) | [sample openj9](https://raw.githubusercontent.com/KoskiLabs/jdk-wrapper/master/examples/adpt.openj9.11.jdkw), [sample hotspot](https://raw.githubusercontent.com/KoskiLabs/jdk-wrapper/master/examples/adpt.hotspot.8.jdkw)
```
Comment (vjkoskela) [inline README.md]: nit: can you rename the files and update the links from `adpt` to `adopt` as well?

**Comment by vjkoskela** [comment]
Commandeered and merged in https://github.com/KoskiLabs/jdk-wrapper/pull/36.

**Pull request merged**